models/articles.model.js

```diff
@@ -81,3 +81,61 @@ exports.fetchCommentsByArticleId = (article_id) => {
       return rows;
     });
 };
+
+const checkIfUserExists = (author) => {
+  return db
+    .query(
+      `
+      SELECT * FROM users 
+      WHERE username = $1
+      `,
+      [author]
+    )
+    .then(({ rows }) => {
+      return rows.length > 0;
+    });
+};
+
+const checkIfArticleExists = (article_id) => {
+  return db
+    .query(
+      `
+      SELECT * FROM articles 
+      WHERE article_id = $1
+      `,
+      [article_id]
+    )
+    .then(({ rows }) => {
+      return rows.length > 0;
+    });
+};
+
+exports.insertComment = ({ article_id, author, body }) => {
+  const checkUser = checkIfUserExists(author);
+  const checkArticle = checkIfArticleExists(article_id);
+
+  return Promise.all([checkUser, checkArticle]).then((data) => {
+    if (!data[0]) {
```

Comment:
This is a smart use of `Promise.all` , however, both _article_id_ and _author_ are foreign keys _REFERENCING_ other tables. So you don't need to do any manual checking that they exist, you can use the one _POST_ query and if they don't exist PSQL will throw an error automatically.
This manual approach you have does also work, so it's not critical, but if you have time it would be good to refactor as it would reduce a lot of code